File: algorithms/20.swift
// https://leetcode.com/problems/valid-parentheses

class Solution {
    func isValid(s: String) -> Bool {
        var stack = [String]()
        let dict = [")":"(", "]":"[", "}":"{"]
        for c in s.characters {
            let cs = String(c)
            if !dict.keys.contains(cs) {
                stack.append(cs)
            } else if stack.count == 0 || dict[cs] != stack.popLast() {
                return false
            }
        }
        return stack.count == 0
    }
}

File: algorithms/20.py
# https://leetcode.com/problems/valid-parentheses


class Solution(object):
    def isValid(self, s):
        """
        :type s: str
        :rtype: bool
        """
        stack = []
        dict = {')': '(', ']': '[', '}': '{'}

        for c in s:
            if c not in dict:
                stack.append(c)
            elif len(stack) == 0 or dict[c] != stack.pop():
                return False
        return len(stack) == 0
